infra/base/terraform/vpc.tf

```diff
@@ -37,8 +37,8 @@ module "vpc" {
   # ------------------------------
   # Private Subnets for MLflow backend store
   database_subnets                   = local.database_private_subnets
-  create_database_subnet_group       = true
-  create_database_subnet_route_table = true
+  create_database_subnet_group       = var.enable_mlflow_tracking
+  create_database_subnet_route_table = var.enable_mlflow_tracking
```

Comment:
I would rename the variable to `var.enable_database_subnets` to keep it generic and set to false by default. Blueprints which requires database_subnet_group (e.g., mlflow) can enable this in tfvars file